plugins/websocket_server/WebsocketServer.cc

```diff
@@ -40,22 +40,152 @@ using namespace ignition::launch;
 /// \return A string that is the frame header.
 #define BUILD_MSG(_op, _topic, _type, _payload) (BUILD_HEADER(_op, _topic, _type) + _payload)
 
-int rootCallback(struct lws *_wsi,
-                 enum lws_callback_reasons _reason,
-                 void * /*user*/,
-                 void * _in,
-                 size_t /*len*/)
-{
+/// \brief Gets the websocket server from the lws protocol passed to handler.
+/// \attention The protocol should be defined with a reference to a websocket
+/// server in the `user` field.
+/// \param[in] _wsi lws connection.
+/// \return A pointer to the websocket server assigned to the protocol.
+WebsocketServer *get_server(struct lws *_wsi) {
   WebsocketServer *self = nullptr;
 
   // Get the protocol definition for this callback
-  lws_protocols *protocol = const_cast<lws_protocols*>(
-      lws_get_protocol(_wsi));
+  lws_protocols *protocol = const_cast<lws_protocols*>(lws_get_protocol(_wsi));
 
   // It's possible that the protocol is null.
   if (protocol)
     self = static_cast<WebsocketServer*>(protocol->user);
 
+  return self;
+}
+
+/// \brief Sets HTTP response status code and writes Content-Type and
+/// Content-Lenght HTTP headers.
+/// \param[in] _wsi lws connection.
+/// \param[in] status_code Status code.
+/// \param[in] content_type Content mime-type.
+/// \param[in] content_length Size of the body in bytes.
+int write_http_headers(struct lws *_wsi,
+                       int status_code,
+                       const char *content_type,
+                       unsigned long content_length) {
+  unsigned char buf[4096 + LWS_PRE];
+  unsigned char *p, *start, *end;
+  int n;
+
+  p = buf + LWS_PRE;
+  start = p;
+  end = p + sizeof(buf) - LWS_PRE;
+
+  // Status code
+  if(lws_add_http_header_status(_wsi,
+      status_code,
+      reinterpret_cast<unsigned char **>(&p),
+      end))
+    return 1;
+
+  // Content-Type
+  if (lws_add_http_header_by_token(_wsi,
+      WSI_TOKEN_HTTP_CONTENT_TYPE,
+      reinterpret_cast<const unsigned char *>(content_type),
+      strlen(content_type),
+      &p,
+      end))
+    return 1;
+
+  // Content-Length
+  if (lws_add_http_header_content_length(_wsi,
+      content_length-1,
+      &p,
+      end))
+    return 1;
+
+  // Finalize header
+  if (lws_finalize_http_header(_wsi, &p, end))
+    return 1;
+
+  // Write headers
+  n = lws_write(_wsi, start, p - start, LWS_WRITE_HTTP_HEADERS);
+  if (n < 0)
+    return 1;
+
+  return 0;
+}
+
+int httpCallback(struct lws *_wsi,
+                 enum lws_callback_reasons _reason,
+                 void * user,
+                 void * _in,
+                 size_t len) {
```

Comment:
```suggestion
                 size_t _len) {
```

Reply:
Addressed in https://github.com/ignitionrobotics/ign-launch/pull/49/commits/1fee42e8a1c08937f8610a1581d5926efdb396f1.